src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ViewBuffer.cs

```diff
@@ -25,6 +25,9 @@ public class ViewBuffer : IHtmlContentBuilder
         private readonly IViewBufferScope _bufferScope;
         private readonly string _name;
         private readonly int _pageSize;
+        private ViewBufferPage _singlePage = null;          // Limits allocation if the ViewBuffer has only one page (frequent case).
```

Comment:
Of course no problem to add new test cases. But I'm sorry to waste your time, I don't understand what case you want to cover.
In my test `Append_CreatesManyPages` I add a new value after `Clear` to check if a new page is correctly created. You want to add more than one page after `Clear`?